doc/howtos/profilecode.rst

```diff
@@ -0,0 +1,88 @@
+===================
+Profiling Odoo code
+===================
+
+.. warning::
+
+    This tutorial requires :ref:`having installed Odoo <setup/install>`
+    and :ref:`writing Odoo code <backend>`
+
+Graph a method
+==============
+
+Odoo embeds a profiler of code. This embeded profiler output can be used to
+generate a graph of calls triggered by the method, percentage of time taken in
+the method itself as well as time taken in method and it's sub-called methods.
+
+.. code:: python
+
+    from odoo.tools.misc import profile
+    [...]
+    @profile('/temp/prof.profile')
+    @api.multi
+    def mymethod(...)
+
+This produce a file called /temp/prof.profile
+
+A tool called *gprof2dot* will produce a graph with this result:
+
+.. code:: bash
+
+    gprof2dot -f pstats -o /temp/prof.xdot /temp/prof.profile
+
+A tool called *xdot* will display the resulting graph:
+
+.. code:: bash
+    
+    xdot /temp/prof.xdot
+    
+Dump stack
+==========
+
+Sending the SIGQUIT signal to an odoo process (only available on POSIX) makes
+this process output the current stack trace to log, with info level. When an
+odoo process seems stucked, sending this signal to the process permit to know
+what the process is doing, and letting the process continue his job.
+
+Tracing code execution
+======================
+
+Instead of sending the SIGQUIT signal to an odoo process often enough, to check
+where processes is performing worse than expected, we can use pyflame tool to
+do it for us.
+
+Install pyflame and flamegraph
+------------------------------
+
+.. code:: bash
+
+    sudo apt install autoconf automake autotools-dev g++ pkg-config python-dev python3-dev libtool make
```

Comment:
or a "install the following dependencies", not everybody is using debian-based

Reply:
It's still pretty weird, some of these packages are either debian-only or the names are different in other package repos.

`autotools-dev` seems to be a hack package to make up for the fact that debian is a dinosaur, this package does not exist in fedora nor arch.

`python(3)-dev` is called python(3)-devel in fedora and doesn't exist in arch (nor in any other rolling-release distro, probably).

There's no easy way to list distro-agnostic dependencies, I suggest we just stick with how it was originally and add that these are Ubuntu/Debian specific instructions, as IMO other distro users should be more comfortable figuring stuff on their own.